### PR TITLE
Features/shared and global modules for user auth

### PR DESCRIPTION
**Pull request by smurph7894**
# Description

Added a global module where global services can be used in any module without importing, exporting in each module file. Other modules can be imported and exported out of this module to become available throughout all modules as well. Also added a global service for querying req.user.userId and teamId as well as switched userId for req were commented for auth.  

#45 

Fixes # (issue)
added user auth were commented in modules

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested via swagger.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/teams/teams.controller.ts [hunk 1]
```diff
@@ -5,10 +5,13 @@ import {
...
     Param,
     ParseIntPipe,
+    Request,
+    UseGuards,
 } from "@nestjs/common";
 import { TeamsService } from "./teams.service";
-import { ApiOperation, ApiTags } from "@nestjs/swagger";
+import { ApiBearerAuth, ApiOperation, ApiTags } from "@nestjs/swagger";
 import { UpdateTeamMemberDto } from "./dto/update-team-member.dto";
+import { JwtAuthGuard } from "src/auth/jwt-auth.guard";
```
Comment (kennytram) [inline src/teams/teams.controller.ts:14]: can you change src/auth/jwt... to ../auth/jwt-auth.guard?

File: src/teams/teams.service.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
 import { Injectable } from "@nestjs/common";
 import { UpdateTeamMemberDto } from "./dto/update-team-member.dto";
 import { PrismaService } from "../prisma/prisma.service";
+import { GlobalService } from "src/global/global.service";
```
Comment (kennytram) [inline src/teams/teams.service.ts:4]: Can you make src/global/global.service to ../global/global.service?

File: src/resources/resources.service.ts [hunk 3]
```diff
@@ -7,10 +7,14 @@ import {
 import { CreateResourceDto } from "./dto/create-resource.dto";
 import { UpdateResourceDto } from "./dto/update-resource.dto";
 import { PrismaService } from "src/prisma/prisma.service";
+import { GlobalService } from "src/global/global.service";
```
Comment (kennytram) [inline src/resources/resources.service.ts:10]: Same about this one in resources.

File: src/ideations/ideations.service.ts [hunk 4]
```diff
@@ -7,27 +7,27 @@ import {
 import { PrismaService } from "src/prisma/prisma.service";
 import { CreateIdeationDto } from "./dto/create-ideation.dto";
 import { UpdateIdeationDto } from "./dto/update-ideation.dto";
-import { CreateIdeationVoteDto } from "./dto/create-ideation-vote.dto";
-import { DeleteIdeationVoteDto } from "./dto/delete-ideation-vote.dto";
-import { DeleteIdeationDto } from "./dto/delete-ideation.dto";
+import { GlobalService } from "src/global/global.service";
```
Comment (kennytram) [inline src/ideations/ideations.service.ts:10]: Same for ideations.

File: src/ideations/ideations.controller.ts [hunk 5]
```diff
@@ -7,17 +7,22 @@ import {
...
-import { DeleteIdeationVoteDto } from "./dto/delete-ideation-vote.dto";
-import { ApiCreatedResponse, ApiOperation, ApiTags } from "@nestjs/swagger";
+import {
+    ApiBearerAuth,
+    ApiCreatedResponse,
+    ApiOperation,
+    ApiTags,
+} from "@nestjs/swagger";
 import { Ideation } from "./entities/ideation.entity";
+import { JwtAuthGuard } from "src/auth/jwt-auth.guard";
```
Comment (kennytram) [inline src/ideations/ideations.controller.ts:23]: Ideations controller too.

File: src/techs/techs.controller.ts [hunk 6]
```diff
@@ -7,12 +7,13 @@ import {
...
+    Request,
+    UseGuards,
 } from "@nestjs/common";
 import { TechsService } from "./techs.service";
-import { ApiOperation, ApiTags } from "@nestjs/swagger";
+import { ApiBearerAuth, ApiOperation, ApiTags } from "@nestjs/swagger";
 import { CreateTeamTechDto } from "./dto/create-tech.dto";
-import { CreateTeamTechVoteDto } from "./dto/create-tech-vote.dto";
-import { DeleteTeamTechVoteDto } from "./dto/delete-tech-vote.dto";
+import { JwtAuthGuard } from "src/auth/jwt-auth.guard";
```
Comment (kennytram) [inline src/techs/techs.controller.ts:16]: Same for techs.

File: src/global/global.service.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { Injectable, NotFoundException } from "@nestjs/common";
+import { PrismaService } from "src/prisma/prisma.service";
```
Comment (kennytram) [inline src/global/global.service.ts:2]: src/prisma/prisma.service to ../src/prisma.service.

File: src/global/global.module.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { Module, Global } from "@nestjs/common";
+import { GlobalService } from "./global.service";
+import { PrismaModule } from "src/prisma/prisma.module";
```
Comment (kennytram) [inline src/global/global.module.ts:3]: this prisma one too.

File: src/features/features.service.ts [hunk 9]
```diff
@@ -6,10 +6,14 @@ import {
 import { PrismaService } from "src/prisma/prisma.service";
 import { CreateFeatureDto } from "./dto/create-feature.dto";
 import { UpdateFeatureDto } from "./dto/update-feature.dto";
+import { GlobalService } from "src/global/global.service";
```
Comment (kennytram) [inline src/features/features.service.ts:9]: same for the one in feature.

**Review by kennytram — APPROVED** [review]
otherwise lgtm.

mb I should've started a review first.

**Review by kennytram — APPROVED** [review]
lgtm

**Review by cherylli — APPROVED** [review]

**Pull request merged**